tools/tm-bench/transacter_test.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// This test tests that the output of generate tx and update tx is consistent
+func TestGenerateTxUpdateTxConsistentency(t *testing.T) {
+	cases := []struct {
+		connIndex        int
+		startingTxNumber int
+		txSize           int
+		hostname         string
+		numTxsToTest     int
+	}{
+		{0, 0, 50, "localhost:26657", 1000},
+		{70, 300, 10000, "localhost:26657", 1000},
+		{0, 50, 100000, "localhost:26657", 1000},
+	}
+
+	for tcIndex, tc := range cases {
+		hostnameHash := md5.Sum([]byte(tc.hostname))
+		// Tx generated from update tx. This is defined outside of the loop, since we have
+		// to a have something initially to update
+		updatedTx := generateTx(tc.connIndex, tc.startingTxNumber, tc.txSize, hostnameHash)
+		updatedHex := make([]byte, len(updatedTx)*2)
+		hex.Encode(updatedHex, updatedTx)
+		for i := 0; i < tc.numTxsToTest; i++ {
+			expectedTx := generateTx(tc.connIndex, tc.startingTxNumber+i, tc.txSize, hostnameHash)
+			expectedHex := make([]byte, len(expectedTx)*2)
+			hex.Encode(expectedHex, expectedTx)
+
+			updateTx(updatedTx, updatedHex, tc.startingTxNumber+i)
+
+			// after first 32 bytes is 8 bytes of time, then purely random bytes
+			require.Equal(t, expectedTx[:32], updatedTx[:32],
+				"First 32 bytes of the txs differed. tc #%d, i #%d", tcIndex, i)
+			require.Equal(t, expectedHex[:64], updatedHex[:64],
+				"First 64 bytes of the hex differed. tc #%d, i #%d", tcIndex, i)
+			// Test the lengths of the txs are as expected
+			require.Equal(t, tc.txSize, len(expectedTx),
+				"Length of expected Tx differed. tc #%d, i #%d", tcIndex, i)
+			require.Equal(t, tc.txSize, len(updatedTx),
+				"Length of expected Tx differed. tc #%d, i #%d", tcIndex, i)
+			require.Equal(t, tc.txSize*2, len(expectedHex),
+				"Length of expected hex differed. tc #%d, i #%d", tcIndex, i)
+			require.Equal(t, tc.txSize*2, len(updatedHex),
+				"Length of updated hex differed. tc #%d, i #%d", tcIndex, i)
+		}
+	}
+}
+
+func BenchmarkIterationOfSendLoop(b *testing.B) {
```

Comment:
this currently has code duplication because we don't yet have an implementation of a mock web socket. (As far as I've been able to find) 

The websocket connection isn't an interface sadly, so making a mock one is non-trivial as far I can discern. We have to initialize the websocket.Dialer to a net.Conn which meets the websocket requirements. (Probably not _too_ hard, but does have to do more than return nil everywhere)